src/reflect/scala/reflect/internal/Names.scala

```diff
@@ -398,11 +398,28 @@ trait Names extends api.Names {
       false
     }
 
+    def lastIndexOf(s: String): Int = {
+      if (s.isEmpty()) return length
+
+      val last = s(s.length - 1)
+      var i = length
+      while (s.length() <= i) {
+        if (_chrs(start + i - 1) == last) {
+          var j = 2
+          while (j <= s.length() && _chrs(start + i - j) == s.charAt(s.length() - j))
+            j += 1
+          if (j > s.length()) return i - s.length()
```

Comment:
Compute `s.length()` once?

Reply:
The real problem is that use of parens is inconsistent.